Source/ARTPush.m

```diff
@@ -120,34 +120,40 @@ - (void)getActivationMachine:(void (^)(ARTPushActivationStateMachine *const))blo
     };
 
     if (_activationMachine == nil) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            // -[UIApplication delegate] is an UI API call, so needs to be called from main thread.
-            id delegate = UIApplication.sharedApplication.delegate;
-            [self createActivationStateMachineWithDelegate:delegate
-                                         completionHandler:^(ARTPushActivationStateMachine *const machine) {
-                callbackWithUnlock(machine);
-            }];
-        });
+        id<ARTPushRegistererDelegate, NSObject> delegate = _rest.options.pushRegistererDelegate;
+        if (_rest && delegate) {
```

Comment:
I think for `delegate` not to be `nil` at this point then `_rest` has got not to be `nil`, right? So...

```suggestion
        if (delegate) {
```